datalake/src/main/java/com/sequenceiq/datalake/flow/verticalscale/diskupdate/handler/DatalakeDiskUpdateValidationHandler.java

```diff
@@ -0,0 +1,119 @@
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.StackV4Endpoint;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.request.DiskUpdateRequest;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response.StackV4Response;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response.resource.ResourceType;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response.resource.ResourceV4Response;
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+import com.sequenceiq.cloudbreak.auth.crn.RegionAwareInternalCrnGeneratorFactory;
+import com.sequenceiq.cloudbreak.cloud.model.Volume;
+import com.sequenceiq.cloudbreak.cloud.model.VolumeSetAttributes;
+import com.sequenceiq.cloudbreak.common.json.Json;
+import com.sequenceiq.cloudbreak.eventbus.Event;
+import com.sequenceiq.cloudbreak.eventbus.Promise;
+import com.sequenceiq.datalake.entity.DatalakeStatusEnum;
+import com.sequenceiq.datalake.flow.verticalscale.diskupdate.event.DatalakeDiskUpdateEvent;
+import com.sequenceiq.datalake.flow.verticalscale.diskupdate.event.DatalakeDiskUpdateFailedEvent;
+import com.sequenceiq.datalake.flow.verticalscale.diskupdate.event.DatalakeDiskUpdateStateSelectors;
+import com.sequenceiq.flow.reactor.api.event.EventSender;
+import com.sequenceiq.flow.reactor.api.handler.EventSenderAwareHandler;
+
+@Component
+public class DatalakeDiskUpdateValidationHandler extends EventSenderAwareHandler<DatalakeDiskUpdateEvent> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DatalakeDiskUpdateValidationHandler.class);
+
+    private static final long WORKSPACE_ID = 0L;
+
+    @Inject
+    private StackV4Endpoint stackV4Endpoint;
+
+    @Inject
+    private RegionAwareInternalCrnGeneratorFactory regionAwareInternalCrnGeneratorFactory;
+
+    @Inject
+    private DiskUpdateEndpoint diskUpdateEndpoint;
+
+    public DatalakeDiskUpdateValidationHandler(EventSender eventSender) {
+        super(eventSender);
+    }
+
+    @Override
+    public String selector() {
+        return DATALAKE_DISK_UPDATE_VALIDATION_HANDLER_EVENT.selector();
+    }
+
+    @Override
+    public void accept(Event<DatalakeDiskUpdateEvent> datalakeDiskUpdateEvent) {
+        LOGGER.debug("In DatalakeDiskUpdateValidationHandler.accept");
+        DatalakeDiskUpdateEvent payload = datalakeDiskUpdateEvent.getData();
+        try {
+            StackV4Response stackV4Response = stackV4Endpoint.getWithResources(WORKSPACE_ID, payload.getClusterName(), Set.of(), payload.getAccountId());
+            DiskUpdateRequest diskUpdateRequest = payload.getDatalakeDiskUpdateRequest();
+            ResourceV4Response resource = stackV4Response.getResources().stream()
+                    .filter(res -> null != res.getInstanceId() && res.getInstanceGroup().equals(diskUpdateRequest.getGroup())
+                            && res.getResourceType().equals(ResourceType.AWS_VOLUMESET)).toList().get(0);
```

Comment:
```
filter(...).
findFirst()
```
then check optional if present. getting first element is error prone if list is empty

Reply:
done